apps/client/src/app/components/admin-market-data/admin-market-data.component.ts

```diff
@@ -480,11 +480,21 @@ export class GfAdminMarketDataComponent
         dialogRef
           .afterClosed()
           .pipe(takeUntil(this.unsubscribeSubject))
-          .subscribe(({ dataSource, symbol } = {}) => {
+          .subscribe(({ dataSource, symbol, isCurrency } = {}) => {
             if (dataSource && symbol) {
               this.adminService
                 .addAssetProfile({ dataSource, symbol })
                 .pipe(
+                  switchMap(() => {
```

Comment:
Could you move this to the `onSubmit` method within the create asset profile dialog component? Then, the currency-related logic is in one place.